src/App.tsx

```diff
@@ -1,39 +1,77 @@
+/* eslint-disable no-console */
 import React from 'react';
 import './App.scss';
+import Clock from './components/Clock';
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type Props = {
+  clockName: string;
+  hasClock: boolean;
+  timerID: number;
+};
+
+type State = {
+  clockName: string;
+  timerID: number;
+  hasClock: boolean;
+};
+
+export class App extends React.Component<{}, State> {
+  state = {
+    clockName: 'Clock-0',
+    timerID: 1,
+    hasClock: true,
+  };
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+  componentDidMount() {
+    window.setInterval(() => {
+      this.setState({ clockName: getRandomName() });
+    }, 3300);
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+    document.addEventListener('contextmenu', (event) => {
+      event.preventDefault();
+      this.setState({ hasClock: false });
+    });
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+    document.addEventListener('click', () => {
+      this.setState({ hasClock: true });
+    });
+  }
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+  componentDidUpdate = (_prevProps: Props, prevState: Props) => {
```

Comment:
I believe, it's gonna work 😄 
```suggestion
  componentDidUpdate = (_: Props, prevState: State) => {
```